rules/macos/process_creation/proc_creation_macos_enable_root_account.yml

```
@@ -0,0 +1,41 @@
+title: Enable the Root Account
+id: 821bcf4d-46c7-4b87-bc57-9509d3ba7c11
+status: experimental
+description: Detects attempts to enable/add an account to the admin group, thus granting the root privilege
+references:
+    - https://github.com/redcanaryco/atomic-red-team/blob/master/atomics/T1078.003/T1078.003.md
+    - https://github.com/elastic/detection-rules/blob/4312d8c9583be524578a14fe6295c3370b9a9307/rules/macos/persistence_enable_root_account.toml
+    - https://ss64.com/osx/dsenableroot.html
+    - https://ss64.com/osx/dseditgroup.html
+    - https://ss64.com/osx/dscl.html
+author: Sohan G (D4rkCiph3r)
+date: 2023/02/18
+tags:
+    - attack.t1078
+    - attack.t1078.001
+    - attack.t1078.003
+    - attack.initial_access
+    - attack.persistence
+logsource:
+    category: process_creation
+    product: macos
+detection:
+    selection_dscl:
+        Image|endswith: '/dscl'
+        CommandLine|contains:
+            - '-append' #appends the user
+            - '/Groups/admin'
+    selection_dseditgroup:
+        Image|endswith: '/dseditgroup'
+        CommandLine|contains|all:
+            - '-a' #name of the record(username)
+            - '-t' #type of the record(usergroup)
+            - 'admin'
+    selection_dsenableroot:
+        Image|endswith: '/dsenableroot'
+    filter:
+        CommandLine|contains: ' -d '
+    condition: (selection_dsenableroot and not filter) or 1 of selection*
```

'selection*' will cover selection_dsenableroot too.
```suggestion
    condition: (selection_root and not filter_root) or 1 of selection_ds_*
```
selection_dscl => selection_ds_cl
selection_dseditgroup => selection_ds_editgroup
selection_dsenableroot => selection_root
filter => filter_root